Source/SwiftLintFramework/Rules/EmptyXCTestMethodRule.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import SourceKittenFramework
+
+public struct EmptyXCTestMethodRule: Rule, OptInRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "empty_xctest_method",
+        name: "Empty XCTest Method",
+        description: "Empty XCTest method should be avoided.",
+        kind: .idiomatic,
+        nonTriggeringExamples: EmptyXCTestMethodRuleExamples.nonTriggeringExamples,
+        triggeringExamples: EmptyXCTestMethodRuleExamples.triggeringExamples
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        return testClasses(in: file).flatMap { violations(in: file, for: $0) }
+    }
+
+    // MARK: - Private
+
+    private func testClasses(in file: File) -> [[String: SourceKitRepresentable]] {
+        return file.structure.dictionary.substructure.filter { dictionary in
+            guard
+                let kind = dictionary.kind,
+                SwiftDeclarationKind(rawValue: kind) == .class else { return false }
+
+            return !dictionary.inheritedTypes.filter { $0 == "XCTestCase" }.isEmpty
+        }
+    }
+
+    private func violations(in file: File,
+                            for dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        return dictionary.substructure.compactMap { subDictionary -> StyleViolation? in
+            guard
+                let kind = subDictionary.kind,
+                let swiftKind = SwiftDeclarationKind(rawValue: kind),
+                SwiftDeclarationKind.functionKinds.contains(swiftKind),
+                let name = subDictionary.name, isXCTestMethod(name),
+                let offset = subDictionary.offset,
```

Comment:
can we also test that the method has no parameters just to be safe?

Reply:
Done. Added here: https://github.com/realm/SwiftLint/pull/2189/commits/7cef22b968c518232cd64d26d032e4392a4b18cc